pkg/operator/quorumguardcleanup/quorumguardcleanupcontroller.go

```diff
@@ -0,0 +1,253 @@
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/cluster-etcd-operator/pkg/operator/operatorclient"
+)
+
+const (
+	oldEtcdGuardDeploymentName    = "etcd-quorum-guard"
+	newGuardLabelSelectorString   = "app=guard"
+	masterNodeLabelSelectorString = "node-role.kubernetes.io/master"
+)
+
+// QuorumGuardCleanupController watches the old EtcdQuorumGuard deployment and removes it
+// when the new staticpod controller guard pods have been rolled out
+type QuorumGuardCleanupController struct {
+	expectedOperatorVersion string
+	operatorClient          v1helpers.OperatorClient
+	kubeClient              kubernetes.Interface
+	podLister               corev1listers.PodLister
+	pdbLister               policyv1listers.PodDisruptionBudgetLister
+	deploymentLister        appsv1listers.DeploymentLister
+	masterNodeLister        corev1listers.NodeLister
+	clusterOperatorLister   configv1listers.ClusterOperatorLister
+	clusterVersionLister    configv1listers.ClusterVersionLister
+	infrastructureInformer  configv1informers.InfrastructureInformer
+}
+
+func NewQuorumGuardCleanupController(
+	operatorVersion string,
+	operatorClient v1helpers.OperatorClient,
+	kubeClient kubernetes.Interface,
+	clusterVersionInformer configv1informers.ClusterVersionInformer,
+	clusterOperatorInformer configv1informers.ClusterOperatorInformer,
+	kubeInformers v1helpers.KubeInformersForNamespaces,
+	masterNodeInformer cache.SharedIndexInformer,
+	infrastructureInformer configv1informers.InfrastructureInformer,
+	eventRecorder events.Recorder,
+) factory.Controller {
+	c := &QuorumGuardCleanupController{
+		expectedOperatorVersion: operatorVersion,
+		operatorClient:          operatorClient,
+		kubeClient:              kubeClient,
+		podLister:               kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Pods().Lister(),
+		pdbLister:               kubeInformers.InformersFor(operatorclient.TargetNamespace).Policy().V1().PodDisruptionBudgets().Lister(),
+		deploymentLister:        kubeInformers.InformersFor(operatorclient.TargetNamespace).Apps().V1().Deployments().Lister(),
+		masterNodeLister:        corev1listers.NewNodeLister(masterNodeInformer.GetIndexer()),
+		clusterOperatorLister:   clusterOperatorInformer.Lister(),
+		clusterVersionLister:    clusterVersionInformer.Lister(),
+		infrastructureInformer:  infrastructureInformer,
+	}
+	return factory.New().ResyncEvery(1*time.Minute).WithInformers(
+		kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Pods().Informer(),
+		kubeInformers.InformersFor(operatorclient.TargetNamespace).Apps().V1().Deployments().Informer(),
+		clusterOperatorInformer.Informer(),
+	).WithSync(c.sync).ToController("QuorumGuardCleanupController", eventRecorder.WithComponentSuffix("quorum-guard-cleanup-controller"))
+}
+
+func (c *QuorumGuardCleanupController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	err := c.cleanupOldQuorumGuard(ctx, syncCtx)
+	if err != nil {
```

Comment:
once the error return handles this by using WithSyncDegradedOnError, you don't need to write it here.